packages/rest/src/coercion/coerce-parameter.ts

```diff
@@ -0,0 +1,121 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/rest
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  SchemaObject,
+  ReferenceObject,
+  isReferenceObject,
+} from '@loopback/openapi-v3-types';
+
+import * as HttpErrors from 'http-errors';
+
+/**
+ * Coerce the http raw data to a JavaScript type data of a parameter
+ * according to its OpenAPI schema specification.
+ *
+ * @param data The raw data get from http request
+ * @param schema The parameter's schema defined in OpenAPI specification
+ */
+export function coerceParameter(
+  data: string,
+  schema?: SchemaObject | ReferenceObject,
+) {
+  // ignore reference schema
+  if (!schema || isReferenceObject(schema)) return data;
```

Comment:
Please add a debug statement to allow app developers to figure out why their valid OpenAPI Schema does not trigger parameter coercion.

```ts
if (/*...*/) {
  debug(/*...*/);
  return data;
}
```